### PR TITLE
Highlight door opener

### PR DESCRIPTION
**Pull request by FluxCapacitor2**
When a wither door is opened, the player that opened the door will have their head highlighted in the Spirit Leap menu until a door is opened by a different player, or the blood door is opened.
This takes priority over other highlights

### CODE REVIEW

File: src/main/kotlin/skytils/skytilsmod/features/impl/handlers/SpiritLeap.kt [hunk 1]
```diff
@@ -84,7 +88,9 @@ class SpiritLeap : PersistentSave(File(Skytils.modDir, "spiritleap.json")) {
                     val scale = 0.9f
                     val scaleReset = 1 / scale
                     GlStateManager.pushMatrix()
-                    if (names.getOrDefault(name, false)) {
+                    if(name == doorOpener) {
```
Comment (My-Name-Is-Jeff) [inline src/main/kotlin/skytils/skytilsmod/features/impl/handlers/SpiritLeap.kt]:
```suggestion
                    if (name == doorOpener) {
```

**Review by My-Name-Is-Jeff — APPROVED** [review]

File: src/main/kotlin/skytils/skytilsmod/features/impl/handlers/SpiritLeap.kt [hunk 2]
```diff
@@ -119,6 +125,13 @@ class SpiritLeap : PersistentSave(File(Skytils.modDir, "spiritleap.json")) {
         }
     }
 
+    @SubscribeEvent
+    fun onChatReceived(event: ClientChatReceivedEvent) {
+        if (!Skytils.config.highlightDoorOpener || !Utils.inDungeons || event.type == 2.toByte()) return
+        doorOpener = if (event.message.formattedText == bloodOpenedString) null
+        else (doorOpenedPattern.find(event.message.unformattedText)?.groups?.get("name")?.value ?: return)
```
Comment (My-Name-Is-Jeff) [inline src/main/kotlin/skytils/skytilsmod/features/impl/handlers/SpiritLeap.kt:132]: not too sure if this works

Comment (FluxCapacitor2) [reply src/main/kotlin/skytils/skytilsmod/features/impl/handlers/SpiritLeap.kt:132]: The only thing that I haven't tested is an unranked player, everything else works fine. Theoretically unranked players should work, but I can't be 100% certain without actually testing it

Comment (My-Name-Is-Jeff) [reply src/main/kotlin/skytils/skytilsmod/features/impl/handlers/SpiritLeap.kt:132]: nons don't matter

Comment (My-Name-Is-Jeff) [reply src/main/kotlin/skytils/skytilsmod/features/impl/handlers/SpiritLeap.kt:132]: seems like it would work though

**Pull request merged**